apiserver/facades/client/action/action.go

```diff
@@ -145,20 +167,23 @@ func (a *ActionAPI) Cancel(ctx context.Context, arg params.Entities) (params.Act
 	response := params.ActionResults{Results: make([]params.ActionResult, len(arg.Entities))}
 
 	for i, entity := range arg.Entities {
-		currentResult := &response.Results[i]
-		currentResult.Action = &params.Action{Tag: entity.Tag}
-		tag, err := names.ParseTag(entity.Tag)
+		actionTag, err := names.ParseActionTag(entity.Tag)
 		if err != nil {
-			currentResult.Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
+			response.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
 			continue
 		}
-		_, ok := tag.(names.ActionTag)
-		if !ok {
-			currentResult.Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
+
+		cancelledAction, err := a.operationService.CancelAction(ctx, coreaction.UUID(actionTag.Id()))
+		if errors.Is(err, operationerrors.ActionNotFound) {
```

Comment:
todo: using `juju/errors`, the package needs to switch over to `juju/internal/errors`.